internal/k8s/entity.go

```diff
@@ -239,3 +252,29 @@ func FilterByMatchesPodTemplateSpec(withPodSpec K8sEntity, entities []K8sEntity)
 func (e K8sEntity) ResourceName() string {
 	return fmt.Sprintf("k8s%s-%s", e.Kind.Kind, e.Name())
 }
+
+func (e K8sEntity) HasName(name string) (bool, error) {
+	metas, err := extractObjectMetas(&e)
```

Comment:
i don't think `extractObjectMetas` is the function you want here. This will give you all the object metas...like if you have a deployment, it will give you the PodTemplateSpec metas. and i think you only want the top-level one?

is there a reason you didn't do `e.Name() == name`?

Reply:
...no, no there isn't 😅